content/blog/learn-tidyverse-ai/index.Rmd

```diff
@@ -0,0 +1,540 @@
+For example, a subsequent prompt like the following fixes this issue:
+
+> Can you redraw the plot without coloring each line by track?
+
+You can see this in action at <https://chatgpt.com/share/678f1dee-4af0-8000-befe-6df0a8d70fba>, in the second prompt.
+Here is the updated code suggested by Chat GPT:
+
+```{r}
+# Create the visualization without coloring by track
+ggplot(billboard_long, aes(x = week, y = rank, group = track)) +
+  geom_line(alpha = 0.3, color = "blue") +
+  scale_y_reverse() +  # Reverse the y-axis since rank 1 is the top position
+  labs(
+    title = "Song Rank vs. Week Number",
+    x = "Week Number",
+    y = "Rank",
+    caption = "Data: Billboard (tidyr package)"
+  ) +
+  theme_minimal()
+```
+
+Note, however, that the only change wasn't omitting the `color = track` aesthetic mapping.
+The `alpha` level is also changed (from 0.6 to 0.3) without a justification for that change and the lines are colored `"blue"`.
+None of these are *bad* or *wrong* choices, but they can be confusing for new learners.
+Similarly, using `theme_minimal()` is not a bad or wrong choice either[^2], but it's not *necessary*, but this might not be obvious to a new learner.
+
+[^2]: In fact, it's my preferred ggplot2 theme!
+
+Furthermore, while Chat GPT "solves" the problem, a thorough code review reveals a number of not-so-great things about the answer that can be confusing for new learners or promote poor coding practices:
+
+-   The code loads packages that are not necessary: tidyr and ggplot2 packages are sufficient for this code, we don't also need dplyr. Additionally, learners coming from R for Data Science likely expect `library(tidyverse)` in analysis code, instead of loading the packages individualy.
+-   There is no need to load the `billboard` dataset, it's available to use once the tidyr package is loaded. Additionally, quotes are not needed, `data(billboard)` also works.
+-   The code mixes up tidyverse and base R styles:
+    -   Changing the type of `week` to numeric can be done in a `mutate()` statement with the tidyverse, which would then warrant loading the dplyr package.
+    -   This can also be done within `pivot_longer()` with the `names_transform` argument.
+
+All of these are addressable with further prompts, as I've done at <https://chatgpt.com/share/678f1dee-4af0-8000-befe-6df0a8d70fba>, in the last two prompts.
+But doing so requires being able to identify these issues and explicitly asking for corrections.
+In practice, I wouldn't have asked Chat GPT to correct everything for me, I would have stopped after the first suggestion, which was a pretty good starting point, and made the improvements myself.
+However, a new learner might assume (and based on my experience seeing lots of near learner code, *does* assume) the first answer is the *right* and *good* or *best* answer since (1) it looks reasonable and (2) it works, sort of.
```

Comment:
```suggestion
However, a new learner might assume (and based on my experience seeing lots of w learner code, *does* assume) the first answer is the *right* and *good* or *best* answer since (1) it looks reasonable and (2) it works, sort of.
```